engine/runtime-compiler/src/main/java/org/enso/compiler/pass/lint/unusedimports/UnusedImportsRemover.java

```diff
@@ -0,0 +1,174 @@
+package org.enso.compiler.pass.lint.unusedimports;
+
+import static scala.jdk.javaapi.CollectionConverters.asJava;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import org.enso.compiler.context.CompilerContext;
+import org.enso.compiler.core.ir.Name;
+import org.enso.compiler.core.ir.Warning.UnusedImport;
+import org.enso.compiler.core.ir.Warning.UnusedSymbolsFromImport;
+import org.enso.compiler.core.ir.module.scope.Import;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Removes unused imports from a module - rewrites the file. Either removes the line with the import
+ * or replaces it with a new one with only the used symbols.
```

Comment:
- desirable effort!
- having _self healing_ linting is better than just reporting issues
- future directions:
    - integration with the IDE
    - official switch - like `--lint=fix`?

Reply:
Tracked by https://github.com/enso-org/enso/issues/13080